connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -105,8 +105,13 @@ public WorkerSourceTask(ConnectorTaskId id,
     }
 
     @Override
-    public void initialize(Map<String, String> config) {
-        this.taskConfig = config;
+    public void initialize(TaskConfig taskConfig) {
+        try {
+            this.taskConfig = taskConfig.originalsStrings();
```

Comment:
The bug that @Ishiihara was looking at was caused by the `ClassCastException` here (or maybe in the connector configuration). Seems like we should have already done that validation before writing the config to storage though... Maybe we can remove the try/catch if we plan to address that problem separately.